KMD.Identity.TestApplications.OpenID.Angular/ClientApp/src/app/app.component.ts

```diff
@@ -11,86 +12,86 @@ import { TestApiCallService } from './test-api-call/test-api-call.service';
+  codeLogin: LoginResponse | undefined = undefined;
+  tokenExchangeLogin: LoginResponse | undefined = undefined;
 
   ngOnInit() {
-    this.oidcSecurityService.checkAuth().subscribe((loginResponse: LoginResponse) => {
-      this.isAuthenticated = loginResponse.isAuthenticated;
-      this.userData = loginResponse.userData;
-      this.accessToken = loginResponse.accessToken;
-      this.idToken = loginResponse.idToken;
-
-      if (!this.isAuthenticated) {
-        const urlParams = new URLSearchParams(window.location.search);
-        const oidcError = urlParams.get('error');
-        const oidcErrorDescription = urlParams.get('error_description');
-
-        if (oidcError || oidcErrorDescription) {
-          this.error = {
-            'Error': oidcError,
-            'Description': oidcErrorDescription
-          }
+    this.errorService.isError$.subscribe(isError => this.isError = isError);
+    this.authenticationContext.codeLogin$.subscribe(
+        loginResponse => {
+          this.codeLogin = loginResponse;
+      });
+
+      this.authenticationContext.tokenExchangeLogin$.subscribe(
+        loginResponse => {
+          this.tokenExchangeLogin = loginResponse;
         }
-      }
-    });
+      )
+    this.InitiateAutoLogInFlowIfConfigured();
+  }
 
-    // A quick fix to initiate the log-in flow immediately from https://test.identity.kmd.dk/
+  requireAuthentication(): boolean {
+    return !(this.codeLogin?.isAuthenticated)
+    && (!this.tokenExchangeLogin?.isAuthenticated)
+  }
+
+  InitiateAutoLogInFlowIfConfigured() {
     if (window.location.search.indexOf('autologin') > -1) {
       this.login();
     }
   }
 
   login() {
-    this.error = null;
-
-    // // todo: This will be removed when approaching completion of the work. For now, this is a developer convenience
-    // // todo: This MUST be removed before merging.
-    // this.domainHint = this.domainHint === "" ? IdentityProviders.KmdAd : this.domainHint;
+    this.showTestApiCall = false;
+    this.showUserDelegation = false;
 
-    this.oidcSecurityService.authorize(ConfigIds.Code, { customParams: { "domain_hint": this.domainHint } });
+    this.authenticationContext.login(ConfigIds.Code, undefined);
   }
 
   logout() {
-    const authOptions = {
-      customParams: {
-        "client_id": this.appConfig.security.clientId,
-      }
-    };
-    this.error = null;
-    this.oidcSecurityService.logoff(ConfigIds.Code, authOptions);
+    this.showTestApiCall = false;
+    this.showUserDelegation = false;
+
+    this.authenticationContext.logout()
   }
   
   callApi() {
     this.showTestApiCall = true;
-    this.testApiCallService.callTestApi(this.accessToken);
+    this.showUserDelegation = false;
+    this.hackForceHideUserDelegation = true;
   }
 
   beginUserDelegation() {
-    this.performUserDelegation = true;
+
+    this.showTestApiCall = false;
+    this.showUserDelegation = true;
+    this.hackForceHideUserDelegation = false;
```

Comment:
Yes I named it this way to make it clear that I HATE this. This should be short lived.

Eventually I intend to remove the two auth profiles, meaning that it will be much simpler to test if we are in a user delegation context or not.